spec/javascripts/fetchSpec.js

```diff
@@ -223,4 +223,51 @@ describe('deferred fetch', function() {
     });
   });
 
+  describe('#ajax', function() {
+    var abort, done, request;
+
+    beforeEach(function() {
+      server.respondWith('GET', '/autocomplete', [200, { "Content-Type": "application/json" }, '[]']);
+    });
+
+    describe('using the abortCallback option', function() {
+      beforeEach(function() {
+        abort = sinon.spy();
+        done  = sinon.spy();
+      });
+
+      it('should call the abortCallback option with an ajax abort function', function() {
+        request = Em.Resource.ajax({
+          url: '/autocomplete',
+          abortCallback: abort
+        });
+
+        expect(abort.called).to.be.true;
+      });
+
+      it('should call the done callback when not aborted', function() {
+        request = Em.Resource.ajax({
+          url: '/autocomplete',
+          abortCallback: abort
+        });
+
+        request.done(done);
+        server.respond();
+        expect(done.called).to.be.true;
+      });
+
+      it('should not call the done callback when abort is called', function() {
+        request = Em.Resource.ajax({
+          url: '/autocomplete',
+          abortCallback: function(abort) {
+            abort();
```

Comment:
do you want to write this in a way where you save off the `abort`, and call it outside of this `ajax` invocation (the way you plan to use this, basically)?